Consolidate block and filesystem tests

jsafrane opened the pull request:
Tests in volumes.go and volumemode.go test very similar thing, each in a slightly different way.

* `should be mountable` in volumes.go tests that a filesystem volume is indeed persistent,  i.e. it writes some data to a volume in a pod and runs a second pod and checks that the data is there.

* `should create sc, pod, pv, and pvc, read/write to the pv, and delete all created resources` in volumemode.go checks that filesystem or block volume can be written and read in a single pod and checks that it got indeed a mounted directory or a block device.

I consolidated them into `should store data` tests - it checks both filesystem and block volumes, it uses two pods to check persistency + it checks that a pod got really a block device or a directory.

/kind cleanup
/sig storage

cc @mkimuram @msau42 

@andyzhangx, I see there are some windows-specific commands in fixture.go, but I did not notice similar windows commands for `CheckVolumeModeOfPath`. Were the original `volumemode.go` tests skipped on Windows?

```release-note
NONE
```

mkimuram commented on test/e2e/storage/testsuites/volumemode.go:
```diff
@@ -196,45 +195,9 @@ func (t *volumeModeTestSuite) defineTests(driver TestDriver, pattern testpattern
 				}()
 				framework.ExpectError(err)
 			})
-		} else {
```
VolumeMode tests now seem to be only for block volume mode tests for block unsupported driver.
So, `testpatterns.FsVolModePreprovisionedPV` and `testpatterns.FsVolModeDynamicPV` in `testPatterns` for the volumeMode tests seem to be unnecessary.
(Then, it would be better to  also remove `VolMode` check and filesystem case.)

jsafrane replied:
I'm leaving the patterns there for now, I have plans for other tests where filesystem will be useful (use FS PV as block in a Pod and expect it to fail and the same with a block PV used as FS in a Pod).

mkimuram replied:
https://github.com/kubernetes/kubernetes/pull/79796 seems useful. So, I agree to leave the patterns there for this commit.

mkimuram commented on test/e2e/storage/testsuites/volumes.go:
```diff
@@ -139,8 +149,11 @@ func (t *volumesTestSuite) defineTests(driver TestDriver, pattern testpatterns.T
 	}
 
-	ginkgo.It("should be mountable", func() {
-		skipPersistenceTest(driver)
```
`skipPersistenceTest` becomes unused now? Then, it would be better to delete it.

jsafrane replied:
Good catch, removed.

mkimuram reviewed (commented):
@jsafrane 

Thank you for working on to consolidate volume tests and normal case of volume mode tests.
They are doing almost the same thing as you point out, so I agree that we should consolidate it.

There are some minor comments.

jsafrane commented on test/e2e/framework/volume/fixtures.go:
```diff
@@ -470,105 +468,128 @@ func TestVolumeClient(client clientset.Interface, config TestConfig, fsGroup *in
+	ginkgo.By("Writing text file contents in the container.")
+	for i, test := range tests {
+		commands := []string{"exec", injectorPod.Name, fmt.Sprintf("--namespace=%v", injectorPod.Namespace), "--"}
+		if test.Mode == v1.PersistentVolumeBlock {
```
Yes, I saw your `sha256sum` usage, but I decided not to use it as I would need to sum the data in the injector pod and parse it from pod output to be able to check it in the reader pod later. `echo` works well here (not sure how well it will work on Windows).

mkimuram commented:
/retest

msau42 commented:
cc @jingxu97 for windows tests

msau42 commented:
How similar are these provisioning tests? https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/provisioning.go#L154

jsafrane committed:
Add block volume support to InjectContent / TestVolumeClient

jsafrane committed:
Refactor TestVolumeClient and InjectContent into common function

These two functions do almost the same thing - one writes data to volumes
in a pod, the second reads them. Let them share the code.

jsafrane committed:
Remove redundant tests

We check that block / filesystem volume is writable in volumes.go

jsafrane committed:
Remove redundant provisioning tests

jsafrane commented:
Looking at /test/e2e/storage/testsuites/provisioning.go, there are two tests that are redundant:

* [should provision storage with defaults](https://github.com/kubernetes/kubernetes/blob/4b5dc0a06d7bf92ac2d50fe72137d36f9d0100c1/test/e2e/storage/testsuites/provisioning.go#L154): it's IMO fully covered by "should store data" test in this PR. **Removed in this PR.**

* [should access volume from different nodes](https://github.com/kubernetes/kubernetes/blob/4b5dc0a06d7bf92ac2d50fe72137d36f9d0100c1/test/e2e/storage/testsuites/provisioning.go#L179) is similar to tests in multivolume.go, but multivolume.go uses two volumes instead of one used in provisioning.go. IMO it's just a detail and I am removing the test from provisioning.go.

* [should provision storage with mount options](https://github.com/kubernetes/kubernetes/blob/4b5dc0a06d7bf92ac2d50fe72137d36f9d0100c1/test/e2e/storage/testsuites/provisioning.go#L164) is **not** covered anywhere else.

* [should provision storage with snapshot data source](https://github.com/kubernetes/kubernetes/blob/4b5dc0a06d7bf92ac2d50fe72137d36f9d0100c1/test/e2e/storage/testsuites/provisioning.go#L202) is **not** covered anywhere else.

msau42 commented:
/lgtm

I checked with @jingxu97, we don't support raw block for windows
